include/nbl/asset/utils/ICPUVirtualTexture.h

```diff
@@ -528,9 +528,38 @@ class ICPUVirtualTexture final : public IVirtualTexture<ICPUImageView, ICPUSampl
         return getDSlayoutBindings_internal<ICPUDescriptorSetLayout>(_outBindings, _outSamplers, _pgtBinding, _fsamplersBinding, _isamplersBinding, _usamplersBinding);
     }
 
-    auto getDescriptorSetWrites(ICPUDescriptorSet::SWriteDescriptorSet* _outWrites, ICPUDescriptorSet::SDescriptorInfo* _outInfo, ICPUDescriptorSet* _dstSet, uint32_t _pgtBinding = 0u, uint32_t _fsamplersBinding = 1u, uint32_t _isamplersBinding = 2u, uint32_t _usamplersBinding = 3u) const
+    bool updateDescriptorSet(ICPUDescriptorSet* _dstSet, uint32_t _pgtBinding = 0u, uint32_t _fsamplersBinding = 1u, uint32_t _isamplersBinding = 2u, uint32_t _usamplersBinding = 3u) const
     {
-        return getDescriptorSetWrites_internal<ICPUDescriptorSet>(_outWrites, _outInfo, _dstSet, _pgtBinding, _fsamplersBinding, _isamplersBinding, _usamplersBinding);
+        // Update _pgtBinding.
+        {
+            auto pgtInfos = _dstSet->getDescriptorInfos(_pgtBinding, IDescriptor::E_TYPE::ET_COMBINED_IMAGE_SAMPLER);
+            if (pgtInfos.empty())
+                return false;
+
+            assert(pgtInfos.size() == 1ull);
```

Comment:
I'd prefer logging and a return false instead of assert

Reply:
`return`-ed `false`, but I don't think we have a logger here, since all of this in the `asset` namespace.

Reply:
`asset` depends on `system` so loggers are possible, but if `ICPUVirtualTexture` doesn't take one already, we can leave the logging as a TODO

Reply:
Added the TODO. Resolved?

Reply:
yes